bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/api/StatusService.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.revogi.internal.api;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.jetbrains.annotations.NotNull;
+import org.openhab.binding.revogi.internal.udp.UdpResponse;
+import org.openhab.binding.revogi.internal.udp.UdpSenderService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link StatusService} contains methods to get a status of a Revogi SmartStrip
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class StatusService {
+
+    private static final String UDP_DISCOVERY_QUERY = "V3{\"sn\":\"%s\", \"cmd\": 90}";
+    public static final String VERSION_STRING = "V3";
+    private final Logger logger = LoggerFactory.getLogger(StatusService.class);
+
+    private final Gson gson = new GsonBuilder().create();
+    private final UdpSenderService udpSenderService;
+
+    public StatusService(UdpSenderService udpSenderService) {
+        this.udpSenderService = udpSenderService;
+    }
+
+    public CompletableFuture<Status> queryStatus(String serialNumber, String ipAddress) {
+        CompletableFuture<List<UdpResponse>> responses;
+        if (ipAddress.trim().isEmpty()) {
+            responses = udpSenderService.broadcastUpdDatagram(String.format(UDP_DISCOVERY_QUERY, serialNumber));
+        } else {
+            responses = udpSenderService.sendMessage(String.format(UDP_DISCOVERY_QUERY, serialNumber), ipAddress);
+        }
+        return responses.thenApply(this::getStatus);
+    }
+
+    @NotNull
+    private Status getStatus(final List<UdpResponse> singleResponse) {
+        singleResponse.forEach(response -> logger.info("Received: {}", response.getAnswer()));
```

Comment:
See above

Reply:
I know what you mean, but on github above and below doesn't seem to work as expected ;)